### PR TITLE
Make test stable

### PR DESCRIPTION
**Pull request by July541**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -27,26 +27,33 @@ jobs:
...
+          export GHCUP_INSTALL_BASE_PREFIX=$(pwd)/test-workspace/bin
+          ghcup install stack latest
+          ghcup install stack recommended
+          ghcup list -t stack -c installed -r
+        shell: bash
       - name: Install cabal
-        run: ghcup install cabal recommended
+        run: |
+          export GHCUP_INSTALL_BASE_PREFIX=$(pwd)/test-workspace/bin
+          ghcup install cabal recommended
```
Comment (hasufell) [inline .github/workflows/test.yml]: not needed

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -27,26 +27,33 @@ jobs:
...
+          ghcup install cabal latest
+          ghcup list -t cabal -c installed -r
+        shell: bash
       - name: Install GHC
         run: |
+          export GHCUP_INSTALL_BASE_PREFIX=$(pwd)/test-workspace/bin
           ghcup install ghc ${{matrix.ghc}}
-          ghcup set ghc ${{matrix.ghc}}
```
Comment (hasufell) [inline .github/workflows/test.yml:36]: why remove setting ghc?

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -27,26 +27,33 @@ jobs:
         shell: bash
       # Setup the environment for the tests
       - name: Install stack
-        run: ghcup install stack recommended
+        run: |
```
Comment (hasufell) [inline .github/workflows/test.yml:32]: this shouldn't be 4 separate steps... do this all in one step where HLS is installed as well

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -56,22 +63,15 @@ jobs:
       - run: yarn run test
```
Comment (hasufell) [inline .github/workflows/test.yml]: Let's export PATH and export GHCUP_INSTALL_BASE_PREFIX here as well:

```sh
export GHCUP_INSTALL_BASE_PREFIX=$(pwd)/test-workspace/bin
export PATH="$(pwd)/test-workspace/bin/bin:$PATH"
```

File: .github/workflows/test.yml (hunk 2 above)
Comment (July541) [reply .github/workflows/test.yml:36]: I just remove this for testing, looks it has no effect.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -25,53 +25,46 @@ jobs:
...
-      - run: |
-          mkdir -p test-workspace/bin/
           export GHCUP_INSTALL_BASE_PREFIX=$(pwd)/test-workspace/bin
-          echo $XDG_BIN_HOME $GHCUP_INSTALL_BASE_PREFIX
           ghcup config set cache true
-          ghcup --no-verbose prefetch hls 1.4.0
-          ghcup --no-verbose prefetch hls latest
-        shell: bash
+          ghcup install stack latest
+          ghcup install stack recommended
```
Comment (hasufell) [inline .github/workflows/test.yml]: don't install recommended stack

File: .github/workflows/test.yml (hunk 1 above)
Comment (hasufell) [reply .github/workflows/test.yml]: drop installing recommended cabal

**Pull request closed without merging**